paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/EditCardDetailsInteractor.kt

```diff
@@ -0,0 +1,95 @@
+package com.stripe.android.paymentsheet.ui
+
+import androidx.compose.runtime.Immutable
+import com.stripe.android.CardBrandFilter
+import com.stripe.android.model.CardBrand
+import com.stripe.android.model.PaymentMethod
+import com.stripe.android.paymentsheet.CardUpdateParams
+import kotlinx.coroutines.flow.StateFlow
+
+internal typealias CardDetailsCallback = (CardUpdateParams?) -> Unit
+
+internal typealias BrandChoiceCallback = (CardBrand) -> Unit
+
+/**
+ * Interface for handling UI interactions when editing card details.
+ */
+internal interface EditCardDetailsInteractor {
+    /**
+     * The card being edited.
+     */
+    val card: PaymentMethod.Card
+
+    /**
+     * Filter for determining which card brands are available.
+     */
+    val cardBrandFilter: CardBrandFilter
+
+    /**
+     * Icon resource ID for the payment method.
+     */
+    val paymentMethodIcon: Int
+
+    /**
+     * Whether to show the card brand dropdown.
+     */
+    val showCardBrandDropdown: Boolean
```

Comment:
imho, `shouldShowCardBrandDropdown` or `isCardBrandDropdownShown` are better names as this is a `Boolean`